subsys/logging/log_backend_native_posix.c

```diff
@@ -91,10 +91,55 @@ static void dropped(const struct log_backend *const backend, u32_t cnt)
 	log_output_dropped_process(&log_output, cnt);
 }
 
+static void sync_string(const struct log_backend *const backend,
```

Comment:
If these functions are only used with LOG_SYNC, shouldn't they be ifdefed accordingly?

Reply:
compiler will take care of that and remove it if LOG_SYNC is not defined because function is not assigned to the struct then:
`.put_sync_string = IS_ENABLED(CONFIG_LOG_SYNC) ?	sync_string : NULL`

Reply:
> compiler will take care of that

That depends on the compiler.. At least I prefer to not relay on too smart compilers. Do I'm not going to complain about this case

Reply:
it's rather a standard compiler feature. Haven't seen compiler that wouldn't do that. It's like:
```
if (0) {
 /*do something */
}
```
It's expected that compiler will remove that if statement, right?